src/Microsoft.Diagnostics.Monitoring/DiagnosticsMonitor.cs

```diff
@@ -127,7 +132,7 @@ public async ValueTask DisposeAsync()
                 _currentTask = null;
                 _disposed = true;
             }
-             _disposeSource.Cancel();
+            _stopProcessingSource.Cancel();
```

Comment:
is it safe to call Cancel twice( once from the public method and once from the dispose?)

Reply:
My understanding is that it is idempotent. You can call it as many times as you want but it will only set the cancellation once on the first call. However, Cancel will throw if the CancellationTokenSource is disposed.